### PR TITLE
Adding ttl property on the paths: pki-backend/issue/role_name and pki-backend/sign/role

### PR DESCRIPTION
**Pull request by angelmoo**
Now user can user ttl when requesting a certificate using the path: pki-backend/issue/role and when sign a certificate using the path pki-backend/sign/role

if the ttl attribute is set on this paths and on the role, the one is set on the path will be used, if is not set on the path but is set on the role, then the one is set on the role will be used.

Also added test case for each one of this new features.

ran test suite and all test passed.

### CODE REVIEW

File: plugin/pki/path_venafi_cert_enroll.go [hunk 1]
```diff
@@ -48,6 +48,13 @@ func pathVenafiCertEnroll(b *backend) *framework.Path {
 				Type:        framework.TypeCommaStringSlice,
 				Description: "Use to specify custom fields in format 'key=value'. Use comma to separate multiple values: 'key1=value1,key2=value2'",
 			},
+			"ttl": {
+				Type: framework.TypeDurationSecond,
+				Description: `The lease duration if no specific lease duration is
+requested. The lease duration controls the expiration
+of certificates issued by this backend. Defaults to
+the value of max_ttl.`,
+			},
```
Comment (tr1ck3r) [inline plugin/pki/path_venafi_cert_enroll.go:57]: This description applies to roles not certificates.  Please change to the following (which is basically the same as the native Vault PKI's path-help description for the `ttl` parameter):
> The requested Time To Live for the certificate; sets the expiration date. If not specified the role default is used. Cannot be larger than the role max TTL.

File: plugin/pki/path_venafi_cert_enroll.go [hunk 2]
```diff
@@ -66,6 +73,13 @@ func pathVenafiCertSign(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: `PEM-format CSR to be signed.`,
 			},
+			"ttl": {
+				Type: framework.TypeDurationSecond,
+				Description: `The lease duration if no specific lease duration is
+requested. The lease duration controls the expiration
+of certificates issued by this backend. Defaults to
+the value of max_ttl.`,
+			},
```
Comment (tr1ck3r) [inline plugin/pki/path_venafi_cert_enroll.go:82]: Same comment as above.  Please change to the following:
> The requested Time To Live for the certificate; sets the expiration date. If not specified the role default is used. Cannot be larger than the role max TTL.

File: plugin/pki/path_venafi_cert_enroll.go [hunk 3]
```diff
@@ -96,6 +110,13 @@ func (b *backend) pathVenafiIssue(ctx context.Context, req *logical.Request, dat
 		return logical.ErrorResponse("role key type \"any\" not allowed for issuing certificates, only signing"), nil
 	}
 
+	ttl := time.Duration(data.Get("ttl").(int)) * time.Second
+	if role.MaxTTL > 0 && ttl > role.MaxTTL {
+
+		return logical.ErrorResponse(errorTextValueMustBeLess), nil
```
Comment (tr1ck3r) [inline plugin/pki/path_venafi_cert_enroll.go:116]: This behavior (error) is inconsistent with the native Vault PKI.  If a `max_ttl` value is specified for the role and a `ttl` value greater than that is specified when requesting the certificate, then the validity of the Venafi enrolled certificate should be the role `max_ttl`.

File: plugin/pki/path_venafi_cert_enroll.go [hunk 4]
```diff
@@ -113,6 +134,13 @@ func (b *backend) pathVenafiSign(ctx context.Context, req *logical.Request, data
 		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", roleName)), nil
 	}
 
+	ttl := time.Duration(data.Get("ttl").(int)) * time.Second
+	if role.MaxTTL > 0 && ttl > role.MaxTTL {
+
+		return logical.ErrorResponse(errorTextValueMustBeLess), nil
```
Comment (tr1ck3r) [inline plugin/pki/path_venafi_cert_enroll.go:140]: Same comment as above.  The validity of the Venafi enrolled certificate should be the role `max_ttl` if this situation exists.

**Review by tr1ck3r — COMMENTED** [review]
@angelmoo Not currently in this PR but the descriptions for `ttl` and `max_ttl` in the role need to be revised since the current implementation does not take into account the sys/config `default_lease_ttl` or `max_lease_ttl`.  I don't think we need to take them into account until we support revocation.  However, it means we should not be talking about leases when discussing the parameters in this enhancement.  So let's change the following to simply say: "The certificate validity if no specific certificate validity is requested."
https://github.com/Venafi/vault-pki-backend-venafi/blob/f993bed416d0325ad2aeadde5bc7672bde3160d2/plugin/pki/path_roles.go#L90-L93
And let's change the following to simply say: "The maximum allowed certificate validity"
https://github.com/Venafi/vault-pki-backend-venafi/blob/f993bed416d0325ad2aeadde5bc7672bde3160d2/plugin/pki/path_roles.go#L101

File: plugin/pki/path_venafi_cert_enroll.go (hunk 1 above)
Comment (angelmoo) [reply plugin/pki/path_venafi_cert_enroll.go:55]: Fixed on the latest commit.

File: plugin/pki/path_venafi_cert_enroll.go (hunk 3 above)
Comment (angelmoo) [reply plugin/pki/path_venafi_cert_enroll.go]: fixed on the latest commit.

File: plugin/pki/path_venafi_cert_enroll.go (hunk 4 above)
Comment (angelmoo) [reply plugin/pki/path_venafi_cert_enroll.go]: fixed on the latest commit

**Review by tr1ck3r — APPROVED** [review]
Looks good @angelmoo 😃

**Review by rvelaVenafi — APPROVED** [review]
Fix looks good.

**Pull request merged**